### PR TITLE
For #5182: Loading experiments on startup is slow, remove Fretboard!!

### PR DESCRIPTION
**Pull request by colintheshots**
This removes Fretboard. The goal is to reduce cold startup costs associated with loading the experiments on the main thread. We currently have two experiments frameworks in use and should only require one.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/ext/Context.kt [hunk 1]
```diff
@@ -64,6 +64,9 @@ fun Context.asFragmentActivity() = (this as? ContextThemeWrapper)?.baseContext a
 fun Context.getPreferenceKey(@StringRes resourceId: Int): String =
     resources.getString(resourceId)
 
+@Suppress("unused")
+fun Context.isInExperiment(@Suppress("UNUSED_PARAMETER") experimentName: String): Boolean = false
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/ext/Context.kt]: Nit: would you mind adding a comment along the lines of `Feature flags are not supported by Experimentor, we are currently looking into another way to handle them. See {issue number}`

File: app/src/main/java/org/mozilla/fenix/Experiments.kt [hunk 2]
```diff
@@ -4,27 +4,8 @@
...
-    val AATestDescriptor = ExperimentDescriptor("AAtest")
-    // application services flag to disable the Firefox Sync syncManager
-    val asFeatureSyncDisabled = ExperimentDescriptor("asFeatureSyncDisabled")
-    // application services flag to disable Firefox Accounts pairing button.
-    val asFeatureFxAPairingDisabled = ExperimentDescriptor("asFeatureFxAPairingDisabled")
-    // application services flag to disable Firefox Accounts WebChannel integration.
-    val asFeatureWebChannelsDisabled = ExperimentDescriptor("asFeatureWebChannelsDisabled")
+    const val asFeatureWebChannelsDisabled = "asFeatureWebChannelsDisabled"
+    const val asFeatureSyncDisabled = "asFeatureSyncDisabled"
+    const val asFeatureFxAPairingDisabled = "asFeatureFxAPairingDisabled"
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/Experiments.kt]: It's worth noting that @colintheshots reached out to the devs who created these flags to let them know they won't be remotely toggleable after this change.

**Comment by mcomella** [comment]
I'm probably good with Severin's review but adding myself for a quick skim.

**Comment by colintheshots** [comment]
> I'm probably good with Severin's review but adding myself for a quick skim when we've confirmed we're removing it.

We now have acknowledgement from the FxA team to remove it. They asked me to backlog an A-C issue for server-side flags, [which I did](https://github.com/mozilla-mobile/android-components/issues/5528).

The only open question is if I should remove the `isInExperiment` function entirely or else add `@Deprecated` flags and `@Suppress` flags to the statements that call them.

**Comment by colintheshots** [comment]
@Baron-Severin and I agreed that it makes more sense to migrate this functionality to `FeatureFlags` rather than keeping a deprecated `isInExperiment` method.

**Comment by colintheshots** [comment]
How can I get it to re-run the UI tests at this point?

**Comment by mcomella** [comment]
> How can I get it to re-run the UI tests at this point?

iirc, we're not using bors so we do the old Focus thing: edit the PR title.

**Comment by mcomella** [comment]
Which I already did.

**Comment by mcomella** [comment]
profile

**Comment by mcomella** [comment]
We may want to land this tomorrow to be able to detect a monitor for possible regressions in other PRs in tonight's build.

**Comment by mcomella** [comment]
CI failure looks like a network error: rerunning.

Note: I did not see any regressions from https://github.com/mozilla-mobile/fenix/pull/7487#issuecomment-573343816 so I'm happy for this to merge. I had wanted to land that potential regression separately from this PR so that any improvements here weren't washed away by a possible known regression.

Reminder from https://github.com/mozilla-mobile/fenix/pull/7510#pullrequestreview-341433483 that we should remove the fretboard dependencies.

**Pull request merged**